chess/ChessGameModel.swift

```diff
@@ -23,17 +23,26 @@ struct ChessGameModel {
     }
 
     mutating func intentionToMovePiece(from source: Coordinate, to destination: Coordinate) throws {
-        guard let piece = self.board[source.row][source.col] else {
+        guard let pieceToMove = self.board[source.row][source.col] else {
             throw ChessGameModelError.missingPiece
         }
 
-        if isPawnPromotionMove(piece: piece, to: destination) {
+        let pieceToEat = self.board[destination.row][destination.col]
+        if pieceToEat != nil {
+            if pieceToEat!.color != pieceToMove.color {
```

Comment:
```suggestion
        if let pieceToEat, pieceToEat!.color != pieceToMove.color {
```
this is how it usually looks in swift codebase, we are unwrapping optional and in the same if statement checking all conditions that we need